src/runaqa.ts

```diff
@@ -1,324 +1,475 @@
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {[null]}  null
+ */
+export async function setupParallelEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string,
+  numMachines: string
+): Promise<void> {
+
+  await setupTestEnv(version, jdksource, customizedSdkUrl, sdkdir, buildList, aqatestsRepo, openj9Repo, tkgRepo, vendorTestParams, aqasystemtestsRepo);
+  process.chdir('TKG');
+  process.env.PARALLEL_OPTIONS = `PARALLEL_OPTIONS=TEST=${buildList} TEST_TIME= NUM_MACHINES=${numMachines}`;
+  await exec.exec(`make genParallelList ${process.env.PARALLEL_OPTIONS}`);
+
+}
+
+/**
+ * Sets up the test environment on the runner.
+ * @param  {string} version JDK Version being tested
+ * @param  {string} jdksource Source for JDK
+ * @param  {[string]} customizedSdkUrl Download link for JDK binaries
+ * @param  {[string]} sdkdir Directory for SDK
+ * @param  {[string]} buildList AQAvit Test suite
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {null}  null
+ */
+async function setupTestEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string
+  ):  Promise<void>{
+    await installPlatformDependencies();
+    setSpec();
+    process.env.BUILD_LIST = buildList;
+    if ((jdksource === 'upstream' ||
+        jdksource === 'github-hosted' ||
+        jdksource === 'install-jdk') &&
+        !('TEST_JDK_HOME' in process.env)) {
+        process.env.TEST_JDK_HOME = getTestJdkHome(version, jdksource);
+    }
+    if (!('TEST_JDK_HOME' in process.env)) {
+        process.env.TEST_JDK_HOME = `${sdkdir}/openjdkbinary/j2sdk-image`;
+    }
+
+    await getAqaTestsRepo(aqatestsRepo, version, buildList);
+    await runGetSh(tkgRepo, openj9Repo, vendorTestParams, jdksource, customizedSdkUrl, sdkdir);
+    resetJDKHomeFromProperties();
+
+    //Get Dependencies, using /*zip*/dependents.zip to avoid loop every available files
+    let dependents = await tc.downloadTool('https://ci.adoptopenjdk.net/view/all/job/test.getDependency/lastSuccessfulBuild/artifact//*zip*/dependents.zip');
+    let sevenzexe = '7z';

```

Comment:
This seems to be a linux specific set up only.  Is that the only support O/S for the AQAVit runner?

Reply:
We have support for Windows/MacOS/Linux (non-ARM).